types/coin.go

```diff
@@ -900,3 +900,75 @@ func ParseCoinsNormalized(coinStr string) (Coins, error) {
 	}
 	return NormalizeCoins(coins), nil
 }
+
+// ----------------------------------------------------------------------------
+// Functions used for testing purposes.
+
+// NormalizeDecCoin try to convert a decimal coin to the smallest unit registered,
+// returns original one if failed.
+func normalizeDecCoin(coin DecCoin) DecCoin {
+	newCoin, err := convertDecCoin(coin, coin.Denom)
+	if err != nil {
+		return coin
+	}
+	return newCoin
+}
+
+// NormalizeCoins normalize and truncate a list of decimal coins
+func NormalizeCoins(coins []DecCoin) Coins {
+	if coins == nil {
+		return nil
+	}
+	result := make([]Coin, 0, len(coins))
+
+	for _, coin := range coins {
+		newCoin, _ := normalizeDecCoin(coin).TruncateDecimal()
+		result = append(result, newCoin)
+	}
+
+	return result
+}
+
+// ConvertDecCoin attempts to convert a decimal coin to a given denomination. If the given
+// denomination is invalid or if neither denomination is registered, an error
+// is returned.
+func convertDecCoin(coin DecCoin, denom string) (DecCoin, error) {
+	if err := ValidateDenom(denom); err != nil {
+		return DecCoin{}, err
+	}
+
+	srcUnit, ok := getDenomUnit(coin.Denom)
+	if !ok {
+		return DecCoin{}, fmt.Errorf("source denom not registered: %s", coin.Denom)
+	}
+
+	dstUnit, ok := getDenomUnit(denom)
+	if !ok {
+		return DecCoin{}, fmt.Errorf("destination denom not registered: %s", denom)
+	}
+
+	if srcUnit.Equal(dstUnit) {
+		return NewDecCoinFromDec(denom, coin.Amount), nil
+	}
+
+	return NewDecCoinFromDec(denom, coin.Amount.Mul(srcUnit).Quo(dstUnit)), nil
+}
+
+// denomUnits contains a mapping of denomination mapped to their respective unit
+// multipliers (e.g. 1atom = 10^-6uatom).
+var denomUnits = map[string]math.LegacyDec{}
```

Comment:
Where is denomUnits being set? I only see it being used to lookup

Reply:
thank you, it was related to the global, it is not needed anymore